.github/scripts/simple-smoke-test.sh

```diff
@@ -29,6 +30,18 @@ function stop_container() {
     docker stop "${container_name}"
 }
 
+function check_java_version() {
+    local expected_major_version=$1
+    echo "Checking Java version - expected ${expected_major_version}"
```

Comment:
nit - move `echo` after all `local` declarations

Reply:
why?

Usually, keeping variable closer to their usage makes the code easier to understand

Reply:
the standard is to declare vars first - at least thats what I do
Helps seeing easily what input vars are needed. 
In other langs this happens implicitly via function declaration
its little distracting otherwise.

Reply:
This is actually another debugging statement I'd left in... removed.

96388d8549078104635f9047ebe655adfe537fe4

Reply:
In this case `actual_major_version` is not an input var so there's no need to move it up and extend its scope. 

> the standard is to declare vars first - at least thats what I do
There's no such standard, maybe in C. 

It's about readability, scope, and maintainability. Take a look at this discussion:
https://softwareengineering.stackexchange.com/questions/113262/why-declare-variables-close-to-where-they-are-used